plugins/jabra-gnp/fu-jabra-gnp-image.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2023 GN Audio A/S
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#include "fu-jabra-gnp-common.h"
+#include "fu-jabra-gnp-image.h"
+
+struct _FuJabraGnpImage {
+	FuArchiveFirmware parent_instance;
+	guint32 crc32;
+};
+
+G_DEFINE_TYPE(FuJabraGnpImage, fu_jabra_gnp_image, FU_TYPE_FIRMWARE)
+
+static void
+fu_jabra_gnp_image_export(FuFirmware *firmware, FuFirmwareExportFlags flags, XbBuilderNode *bn)
+{
+	FuJabraGnpImage *self = FU_JABRA_GNP_IMAGE(firmware);
+	fu_xmlb_builder_insert_kx(bn, "crc32", self->crc32);
+}
+
+gboolean
+fu_jabra_gnp_image_parse(FuJabraGnpImage *self,
+			 XbNode *n,
+			 FuFirmware *firmware_archive,
+			 GError **error)
+{
+	const gchar *crc_str = NULL;
+	const gchar *language;
+	const gchar *name;
+	const gchar *part_str = NULL;
+	guint64 crc_expected = 0;
+	guint64 partition = 0;
+	g_autoptr(FuFirmware) img_archive = NULL;
+	g_autoptr(GBytes) blob = NULL;
+
+	/* only match on US English */
+	language = xb_node_query_text(n, "language", NULL);
```

Comment:
Is there any technical difference in update sequence or firmware format related to language?
Or it is related only to "partition", "crc", "name" node names below?

If I correctly got the idea -- in the ZIP archive might be several firmwares aimed to different languages (?) and only English variant could be used for flashing. Is that a correct assumption?

Reply:
Yes, there are multiple languages in the archive. The thing is; fwupd does know the users locale -- do we don't *have* to flash English. I'm also guessing we don't want to update a device from *1.2.3 French* to *1.2.4 English* so maybe we can query the device and add the current device language as an instance ID?

Reply:
We had a brief talk internally about this. I will follow-up and see what final solution we would like. If I have to query the device for it's language I will need to make a bit of a change to the code, including the parsing.

Reply:
If it helps, we could do the English-only update as a first cut -- but I do think we want to query the current language and match on do avoid *changing* the users language.